website/content/docs/discovery/dns.mdx

```diff
@@ -285,6 +285,39 @@ caching of DNS results. However, there are many situations in which caching is
 desirable for performance and scalability. This is discussed more in the tutorial
 for [DNS caching](https://learn.hashicorp.com/tutorials/consul/dns-caching).
 
+## DNS with ACLs
+
+After enabling [Access Control Lists (ACLs)](/docs/security/acl/acl-system),
+DNS requests will fail with error `NXDOMAIN` unless ACL tokens have been
+configured with the necessary rules.
+
+When a DNS request is made to a Consul agent, the agent will use a previously
+configured token with the request (since there is no means for the initial DNS
+request to specify a token):
+
+- The agent's [`default` token](/docs/agent/options#acl_tokens_default)
+  token, if configured.
+- Otherwise, the built-in [`anonymous` token](/docs/security/acl/acl-system#builtin-tokens).
+  Because the anonymous token is used when any request is made to Consul without
+  explicitly specifying a token, production deployments should not apply policies
+  needed for DNS to this token.
+
+Consul will either authorize or revoke the request, depending on the token's
+policy rules:
+
+- [node](/docs/agent/acl-rules.html#node-rules) read access is
+  required on the node serving DNS requests
```

Comment:
I took a quick look at `dns.go` to see if I could answer this question for myself, but it wasn't immediately obvious, so I thought I'd ask:

The ACL permissions are checked by the server that receives the RPC request, right? 

Why does the token need `node:read` on the node _serving_  (handling?) the DNS request? I would expect it to need `node:read` for any node in the response, which is the "node being queried", not the node handling the request.

Reply:
I'm the wrong person to answer that question. I was trying to make the documentation around which policies are needed clearer than what exists today, which is just:
> Depending on your use case, the token used for DNS may need policy rules for nodes, services, and prepared queries. 

... which is not very helpful. I wrote that clarification based on what I could gather from readings lots of docs. I haven't had enough practice tracing through the RPC code yet to understand what's going on fully.

Do you feel like you understand what's happening with reasonable confidence? Here are a few questions we should be able to answer:
- For service lookups, do you need node permissions for the nodes on which the service resides? Or just service permissions?
- For node lookups, I assume you need `node:read` on the node being looked up.
- For [connect](https://www.consul.io/docs/discovery/dns#connect-capable-service-lookups) and [ingress](https://www.consul.io/docs/discovery/dns#ingress-service-lookups) lookups ... do you need the same rules as for standard service lookups?

Reply:
All good questions. I think I would be able to answer them, but I'll need to dig through the RPC code a bit to do so. If no one else has the answer to this readily available I'll try and take some time soon to do that digging.

I suspect at the very least this should say something along the lines of:

```suggestion
- [node](/docs/agent/acl-rules.html#node-rules) read access is
  required for any nodes that may be returned in DNS responses, which may be all nodes.
```

As for connect/ingress lookups, I believe those also only use `service:read`.

Reply:
> required for any nodes that may be returned in DNS responses

Does that mean just for any `*.node.consul` lookups? Or also any nodes which host services to be included in `*.service.consul` lookups?

Reply:
> Does that mean just for any *.node.consul lookups? Or also any nodes which host services to be included in *.service.consul lookups?

This applies to explicit node address lookups (`*.node.consul`) and service lookups (`*.service.consul`). The docs for the [`Address`](https://www.consul.io/api-docs/agent/service#address) parameter state:

> Specifies the address of the service. If not provided, the agent's address is used as the address for the service during DNS queries.

The token used for the initial request must have `node:read` permissions for the node(s) on which the service is running in order to be able to resolve and return the agent/node IP address.

Reply:
Thanks @blake and @dnephin. I've updated the wording given the information you provided. I also restructured this list to be organized by type of lookup rather than by type of permission, as that felt easier to understand.